src/protocol.rs

```diff
@@ -339,9 +341,34 @@ impl ParticipantIdentifier {
     }
 }
 
-impl ProofContext for Vec<ParticipantIdentifier> {
+pub struct ProtocolContext {
+    pub participants: Vec<ParticipantIdentifier>,
+    pub generator: CurvePoint,
+    pub order: BigNumber,
+}
+impl ProofContext for ProtocolContext {
     fn as_bytes(&self) -> Vec<u8> {
-        self.iter().flat_map(|pid| pid.0.to_le_bytes()).collect()
+        [
+            self.participants
+                .iter()
+                .flat_map(|pid| pid.0.to_le_bytes())
+                .collect(),
+            serialize!(&self.generator).unwrap(),
+            serialize!(&self.order).unwrap(),
```

Comment:
todo: I want to reduce our use of the `serialize!` macro as much as possible -- it's opaque and ideally we'll remove it from the library entirely (#152). Can you try to rewrite these using `to_bytes` kinds of methods instead? Or maybe even a direct call to `bincode`.